src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -43,11 +39,8 @@ bool cache
 			}
 		}
 
-		if( cache ) {
-			if( string.IsNullOrEmpty( account ) || string.IsNullOrEmpty( role ) ) {
-				throw new BmxException( "Account & Role must be provided when using cached AWS credentials" );
-			}
-
+		// if using cache, avoid calling Okta at all if possible
+		if( cache && !string.IsNullOrEmpty( account ) && !string.IsNullOrEmpty( role ) ) {
```

Comment:
Real change part 1 of 2:

Don't fail the run if account & role aren't provided via config or CLI.

* In my testing, I find erroring out here to be counter intuitive and unnecessarily harsh. It's perfectly fine if users want to get the cached creds early in an interactive run (which will call Okta), and later utilize the cached creds in a script (where account & role are provided via flags and Okta won't be called).
* Erroring out here kinda mixes the purpose of the `cache` and `nonInteractive` parameters of this method, which makes extending this method to work with writing credential process command ([this propsal](https://d2l.slack.com/archives/G4P099831/p1705333949277619)) more difficult.